crates/y-sweet-worker/src/lib.rs

```diff
@@ -84,12 +84,22 @@ async fn new_doc_handler(req: Request, ctx: RouteContext<ServerContext>) -> Resu
 }
 
 async fn new_doc(
-    req: Request,
+    mut req: Request,
     mut ctx: RouteContext<ServerContext>,
 ) -> std::result::Result<NewDocResponse, Error> {
     check_server_token(&req, ctx.data.auth()?)?;
 
-    let doc_id = nanoid::nanoid!();
+    let body = req
+        .json::<DocCreationRequest>()
+        .await
+        .map_err(|_| Error::BadRequest)?;
```

Comment:
Will a request fail if an empty object isn't sent in the body?

Reply:
Yes.